src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -124,12 +126,18 @@ private function transformLinkStatus(PropertyMetadata $propertyMetadata, array $
             $relatedClass = $this->resourceClassResolver->getResourceClass(null, $relatedClass);
         }
 
-        $relatedGroups = $this->getClassSerializerGroups($relatedClass);
+        $relatedGroups = $this->getClassSerializerGroups(
+            $this->getOutputClass($relatedClass) ?? $relatedClass
```

Comment:
Hmm... There is a problem here. `getOutputClass` / `getInputClass` do not respect overrides being set at the operation level. @soyuka Is there a reason why we're not using `getOperationAttribute` instead?

Reply:
Do we really care about operations of the related class here? The operation that matters is the one that led to seriliazing the root object, not the embedded one (`$relatedClass`), which means we only care about whether the embedded object has a global input/output representation, am I right?